### PR TITLE
Feat: --prompt --stream for Bedrock and noop engines.

### LINKED ISSUE #71: Offer streaming responses

### PR DESCRIPTION
**Pull request by DavidSouther**
Streaming responses in the CLI, now the "long" part is waiting for first response which can take a bit for larger contexts.

Demo:

https://github.com/DavidSouther/ailly/assets/498712/5e7db272-6563-41d0-9ee5-8f16a4de2ab2

Closes #71
Groundwork for #65

### CODE REVIEW

**Comment by DavidSouther** [comment]
I can break this into several PRs, but each commit is self-contained.

File: core/src/actions/prompt_thread.test.ts [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+  FileSystem,
+  ObjectFileSystemAdapter,
+} from "@davidsouther/jiffies/lib/esm/fs";
+import { getPlugin, makePipelineSettings } from "../ailly";
+import { getEngine } from "../engine";
+import { LEVEL } from "@davidsouther/jiffies/lib/esm/log";
+import { TIMEOUT } from "../engine/noop";
+
+describe("scheduler", () => {
+  it("limits outstanding tasks", async () => {
```
Comment (cpyle0819) [inline core/src/actions/prompt_thread.test.ts:17]: 👍

File: core/src/util.ts [hunk 2]
```diff
@@ -4,3 +4,15 @@ export const LOGGER = getLogger("@ailly/core");
 LOGGER.level = getLogLevel(process.env["AILLY_LOG_LEVEL"]);
 
 export const isDefined = <T>(t: T | undefined): t is T => t !== undefined;
+
+Promise.withResolvers =
+  Promise.withResolvers ??
```
Comment (cpyle0819) [inline core/src/util.ts:9]: Looks like something Jiffies would have.

**Review by cpyle0819 — APPROVED** [review]

File: cli/index.js [hunk 3]
```diff
@@ -49,15 +49,22 @@ async function main() {
...
-      if (doneSummary.errors) {
-        LOGGER.warn(`Finished with errors`, { errors: doneSummary.errors });
+      if (!args.values.stream) {
+        await finish(generator);
       }
       if (last == "/dev/stdout") {
         const prompt = loaded.context[last];
+        if (args.values.stream) {
+          // Lazy spin until the request starts
+          while (prompt.responseStream == undefined) {
```
[the remark this answers is not in the document]
Comment (DavidSouther) [reply cli/index.js:59]: Correct, await lets the scheduler run other tasks (importantly, the one that fills in responseStream from the bedrock call)

File: core/src/util.ts (hunk 2 above)
Comment (DavidSouther) [reply core/src/util.ts:9]: 2021, yes. This landed in Node 18; jiffies targets 20+. This is only in Ailly for AL2 platforms on Node16.

**Pull request merged**